packages/forms/src/elements/common/Label.tsx

```diff
@@ -71,6 +71,32 @@ export const Label = React.forwardRef<HTMLLabelElement, ILabelProps>((props, ref
       </StyledRadioLabel>
     );
   } else if (type === 'checkbox') {
+    /**
+     * `onLabelSelect` is a workaround for checkbox label `shift + click` bug in Firefox
+     * See: https://bugzilla.mozilla.org/show_bug.cgi?id=559506
+     */
+    const onLabelSelect = (e: React.KeyboardEvent) => {
+      const isFirefox = navigator.userAgent.toLowerCase().indexOf('firefox') > -1;
+
+      if (fieldContext && isFirefox) {
+        const inputProps = fieldContext.getInputProps();
+        const input = document.getElementById(inputProps.id as string) as HTMLInputElement;
```

Comment:
Rather than call `getInputProps()` for the ID, is it possible to retrieve the ID from the `htmlFor` attribute on the current label? Should be available under `e.target`

Reply:
I've updated this to retrieve the ID from the `htmlFor` attribute. This is great, thanks for the suggestion. 

Along with this change, I had to set `pointer-events: none;` to the label's children SVG elements. Otherwise, `onLabelSelect` would be fired for SVG elements and not the label element.